src/main/java/net/atos/entng/rss/helpers/PreferenceHelper.java

```diff
@@ -0,0 +1,54 @@
+package net.atos.entng.rss.helpers;
+
+import fr.wseduc.webutils.Either;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import net.atos.entng.rss.constants.Field;
+import org.entcore.common.neo4j.Neo4j;
+import org.entcore.common.neo4j.Neo4jResult;
+
+public class PreferenceHelper {
+    private static final Neo4j neo = Neo4j.getInstance();
+
+    public static void getPreferences(String userId, Handler<Either<String, JsonArray>> handler) {
+        String query = String.format("MATCH (u:User {id:\"%s\"})-[:PREFERS]->(uac:UserAppConf)"
+                + " RETURN uac.rss LIMIT 1", userId);
+
+        neo.execute(query, new JsonObject().put(Field.USER_ID, userId), Neo4jResult.validUniqueResultHandler(result -> {
+            if (result.isLeft()) {
+                handler.handle(new Either.Left<>(result.left().getValue()));
+                return;
+            }
+            JsonArray pref;
+            String rss = result.right().getValue().getString(Field.PREFERENCES_RSS);
```

Comment:
```suggestion
            handler.handle(new JsonObject(result.right().getValue().getString(Field.PREFERENCES_RSS, "{}")).getJsonArray(Field.PREFERENCES_NO_DISPLAY_ID, new JsonArray))
```

Ici, je t'ais transformé 9 lignes en 1 seule.